linera-witty-macros/src/unit_tests/specialization.rs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Unit tests for the `specialize` attribute.
+
+use super::{super::apply_specialization_attribute, Specialization, Specializations};
+use proc_macro2::Span;
+use quote::{quote, ToTokens};
+use syn::{parse_quote, DeriveInput, Ident};
+
+/// Check that the [`DeriveInput`] of a `struct` is changed.
+#[test]
+fn derive_input_changes() {
+    let mut input: DeriveInput = parse_quote! {
+        #[witty(specialize(First = u8, Second = Vec<bool>))]
```

Comment:
To be sure this comes in addition to the `derive(..)` macro, right?

nit: What do you think of `specialize_with`, `instantiate_with`, `with_types`, or just `with`?
Also, do we need two levels as opposed to `#[witty_specialize_with(First = u8)]`?
E.g. https://docs.rs/serde_with/latest/serde_with/

Reply:
Yep, but for the test the `#[derive(...)]` isn't necessary.

Good idea! I think `witty_specialize_with` is clear and also simplifies the code a bit :tada: Thanks!